src/include/sof/math/trig.h

```diff
@@ -137,17 +156,139 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
  */
 static inline int16_t cos_fixed_16b(int32_t th_rad_fxp)
 {
+	int32_t th_cdc_fxp;
 	int32_t sign;
 	int32_t b_yn;
 	int32_t xn;
-	int32_t th_cdc_fxp;
-	cordic_cfg type = EN_16B_CORDIC_COSINE;
-	/* compute coeff from angles*/
-	cordic_sin_cos(th_rad_fxp, type, &sign, &b_yn, &xn, &th_cdc_fxp);
+
+	cordic_approx(th_rad_fxp, CORDIC_15B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+
 	th_cdc_fxp = sign * xn;
 	/*convert Q1.31 to Q1.15 format*/
 	return sat_int16(Q_SHIFT_RND((sat_int32(Q_SHIFT_LEFT((int64_t)th_cdc_fxp, 30, 31))),
-	31, 15));
+				    31, 15));
+}
+
+/**
+ * CORDIC-based approximation of complex exponential e^(j*THETA).
+ * computes COS(THETA) + j*SIN(THETA) using CORDIC algorithm
+ * approximation and returns the complex result.
+ * THETA values must be in the range [-2*pi, 2*pi). The cordic
+ * exponential algorithm converges, when the angle is in the
+ * range [-pi/2, pi/2).If an angle is outside of this range,
+ * then a multiple of pi/2 is added or subtracted from the
+ * angle until it is within the range [-pi/2,pi/2).Start
+ * with the angle in the range [-2*pi, 2*pi) and output has
+ * range in [-1.0 to 1.0]
+ * +------------------+-----------------+--------+------------+
+ * | thRadFxp	      |cdccexpth        |thRadFxp|   cdccexpth|
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * | 32 | 28  |  1    | 32 | 15 |   1	| 4.28	 | 2.30       |
+ * +------------------+-----------------+--------+------------+
+ */
+static inline struct cmpx cmpx_exp_32b(int32_t th_rad_fxp)
```

Comment:
Thanks. I would like to know how do we replace the legacy exp_fixed(in:Q5.27, out:Q12.20) by cmpx_exp_32b(in:Q4.28, out:Q2.30)

Reply:
At the moment, exponential function [ `cmpx_exp_32b/cmpx_exp_16b` ] will not replace [ `exp_fixed` ] function

Reply:
I have an idea. Could we move exp_fixed and db2lin_fixed function bodies from math/decibels.c to decibels.h, and make them as inline functions? I think it would reduce some cycles for DRC by avoiding frequent jump-calls, because DRC would call those functions several times per frame.

Reply:
Uh I just saw singalsu@'s comment above for inline function disadvantage. That makes senses. Please ignore my comment lol